src/main/java/com/google/api/codegen/transformer/java/JavaGapicSurfaceTransformer.java

```diff
@@ -373,10 +375,20 @@ private StaticLangSettingsView generateSettingsClass(
     xsettingsClass.hasDefaultServiceAddress(interfaceConfig.hasDefaultServiceAddress());
     xsettingsClass.hasDefaultServiceScopes(interfaceConfig.hasDefaultServiceScopes());
     xsettingsClass.hasDefaultInstance(interfaceConfig.hasDefaultInstance());
+    xsettingsClass.generatorVersion(getGeneratorVersion());
 
     return xsettingsClass.build();
   }
 
+  private static String getGeneratorVersion() {
+    // TODO: Read version information from static file.
+    String version = JavaGapicSurfaceTransformer.class.getPackage().getImplementationVersion();
```

Comment:
It's not that much code - you should be able to pull the relevant stuff out of https://github.com/googleapis/gax-java/pull/182 for adding version.txt, populating the resource file in the build output, and reading the resource file at runtime. You wouldn't need to do the other version updating stuff for now.